src/kernel/default.ts

```diff
@@ -644,8 +644,10 @@ class DefaultKernel implements Kernel.IKernel {
     let partialUrl = utils.urlPathJoin(this._wsUrl, KERNEL_SERVICE_URL,
                                        encodeURIComponent(this._id));
     // Strip any authentication from the display string.
-    let parsed = utils.urlParse(partialUrl);
-    let display = partialUrl.replace(parsed.auth, '');
+    if (typeof document !== 'undefined') {
```

Comment:
Since this check is already done in the parse function, instead of relying on the client to *know* how it was implemented, maybe it should instead be:
```typescript
try {
  let parsed = utils.urlParse(partialUrl);
  console.log('Starting websocket', parsed.hostname);
} catch (error) { /* Ignore error. */ }
```